chatwoot/chatwoot
# feat: support reply to for incoming messages on facebook

This PR enables support for "Reply To" on incoming messages on Facebook. This has the following changes

1. Add `ensure_in_reply_to` to ensure that both `in_reply_to_external_id` and `in_reply_to` are set, if one of the values is present
2. Update facebook message parser, and message builder to handle `in_reply_to`

![CleanShot 2023-10-13 at 10 16 50](https://github.com/chatwoot/chatwoot/assets/18097732/b305609d-bc20-4b5a-88a0-e8829792fcf2)

P.S. Facebook only supports reply-to for incoming messages, not for outgoing messages

### How to test

1. Send a message "A" from Facebook to Chatwoot 
2. Send message "B" from Chatwoot to Facebook on that conversation
3. Reply to message "B" with a new message from Facebook, the UI should show up on Chatwoot
4. Reply to the original message "A" with a new message from Facebook again, the UI should show up on Chatwoot
